### PR TITLE
Added a new Servlet to handle requests to the public feed page

### PR DESCRIPTION
**Pull request by Francisco-Ibarra07**
This pull request is part of the work to create a "Public Feed" section. This Feed will display a list of messages that were posted by multiple users. 

Summary of this PR:

- Created a new Servlet called 'MessageFeedServlet'. When you request from the page 'http://localhost:8080/feed', this Servlet simply returns an HTML page with a short message (just for testing :) )

- Added 1 new method to the Datastore class. Similar to the starter code method 'getMessages()', this new method also queries the database but it grabs a collection of messages across multiple users instead of just one specific user.

### CODE REVIEW

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 1]
```diff
@@ -79,4 +79,42 @@ public void storeMessage(Message message) {
...
+  
+      // A Message object is constructed from a user's UUID, user's email, their message, and a timestamp
+      Message message = new Message(id, user, text, timestamp);
+      messages.add(message);
+      
+     } catch (Exception e) {
+      System.err.println("Error reading message.");
+      System.err.println(entity.toString());
+      e.printStackTrace();
+     }
```
Comment (zliou-codeu) [inline src/main/java/com/google/codeu/data/Datastore.java]: Please make the indentation of this block consistent with the rest of the file. In particular, this file uses an indentation width of 2 spaces, but these for-loop and try-catch blocks are indented with one space.

File: src/main/java/com/google/codeu/servlets/MessageFeedServlet.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ * Handles fetching all messages for the public feed.
+ */
+@WebServlet("/feed")
+public class MessageFeedServlet extends HttpServlet {
+  
+ @Override
+ public void doGet(HttpServletRequest request, HttpServletResponse response)
+   throws IOException {
+  
+  response.getOutputStream().println("this will be my message feed");
```
Comment (zliou-codeu) [inline src/main/java/com/google/codeu/servlets/MessageFeedServlet.java]: Please fix this indentation as well. (Should be 2 spaces, but there is only 1 here)

File: src/main/java/com/google/codeu/servlets/MessageFeedServlet.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+/**
+ * Handles fetching all messages for the public feed.
+ */
+@WebServlet("/feed")
+public class MessageFeedServlet extends HttpServlet {
+  
+ @Override
+ public void doGet(HttpServletRequest request, HttpServletResponse response)
+   throws IOException {
```
Comment (zliou-codeu) [inline src/main/java/com/google/codeu/servlets/MessageFeedServlet.java]: This is a continuation of the previous line (aka a line wrap), so it should be indented twice, but there is only one indent here.

This is a style convention for line wrapping, which can vary from project to project and company to company, but generally, double-indenting a line wrap helps make it clearer.
https://google.github.io/styleguide/javaguide.html#s4.5-line-wrapping

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 4]
```diff
@@ -79,4 +79,42 @@ public void storeMessage(Message message) {
...
+   * @return a list of messages posted by multiple users. This list is the public feed
+   */
+  public List<Message> getAllMessages(){
+
+    // Initialize a list to store all messages
+    List<Message> messages = new ArrayList<>();
+  
+    // Query the database for a list of Entities
+    Query query = new Query("Message")
+      .addSort("timestamp", SortDirection.DESCENDING);
```
Comment (zliou-codeu) [inline src/main/java/com/google/codeu/data/Datastore.java]: Line wrap: use 2 indents (see other comment about line wraps)

**Review by zliou-codeu — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/codeu/data/Datastore.java (hunk 1 above)
Comment (Francisco-Ibarra07) [reply src/main/java/com/google/codeu/data/Datastore.java]: Fixed!

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 5]
```diff
@@ -70,7 +70,45 @@ public void storeMessage(Message message) {
 
         Message message = new Message(id, user, text, timestamp);
         messages.add(message);
-      } catch (Exception e) {
+      }
```
Comment (zliou-codeu) [inline src/main/java/com/google/codeu/data/Datastore.java]: Small nit-pick, but usually, the catch keyword is usually on the same line as the closing brace for the try block:
```
try {
  ...
} catch (Exception e) {
  ...
}
```

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 6]
```diff
@@ -70,7 +70,45 @@ public void storeMessage(Message message) {
...
+        String idString = entity.getKey().getName();
+        UUID id = UUID.fromString(idString);
+        String user = (String) entity.getProperty("user");
+        String text = (String) entity.getProperty("text");
+        long timestamp = (long) entity.getProperty("timestamp");
+
+        // A Message object is constructed from a user's UUID, user's email, their message, and a timestamp
+        Message message = new Message(id, user, text, timestamp);
+        messages.add(message);
+      }
```
Comment (zliou-codeu) [inline src/main/java/com/google/codeu/data/Datastore.java]: Same as above comment.

**Review by zliou-codeu — APPROVED** [review]

File: src/main/java/com/google/codeu/data/Datastore.java (hunk 5 above)
Comment (Francisco-Ibarra07) [reply src/main/java/com/google/codeu/data/Datastore.java]: Ok got it!

**Pull request merged**